src/app/dataset-view/additional-components/flows-component/components/flows-table/flows-table.component.ts

```diff
@@ -128,7 +140,12 @@ export class FlowsTableComponent implements OnInit, OnChanges {
         return DatasetFlowTableHelpers.waitingBlockText(startCondition);
     }
 
-    public cancelFlow(flowId: string): void {
+    public datasetById(datasetId: string): Dataset {
+        const dataset = this.datasetFlowOwners?.find((dataset) => dataset.id === datasetId) as Dataset;
+        return dataset;
+    }
```

Comment:
The result might be undefined. It's not reflected in the type

Reply:
Interesting question - why we allow `datasetFlowOwners` to be undefined. It seems to be defined always.

Reply:
`datasetFlowOwners`  must always be defined